### PR TITLE
heartbeat: rebind stream.

### PR DESCRIPTION
**Pull request by disksing**
Circumvent potential problems that may be occurred by reconnecting.

### CODE REVIEW

File: server/grpc_service.go [hunk 1]
```diff
@@ -311,9 +313,10 @@ func (s *Server) RegionHeartbeat(stream pdpb.PD_RegionHeartbeatServer) error {
 
 		hbStreams := cluster.coordinator.hbStreams
 
-		if isNew {
+		if time.Since(lastBind) > heartbeatStreamRebindInterval {
+			regionHeartbeatCounter.WithLabelValues(storeLabel, "report", "bind").Inc()
 			hbStreams.bindStream(storeID, server)
```
Comment (siddontang) [inline server/grpc_service.go]: any side effect if you bindStream again after the interval?

Comment (disksing) [reply server/grpc_service.go]: Just re-assign a map entry, should be trivial.

File: server/grpc_service.go [hunk 2]
```diff
@@ -239,6 +239,8 @@ func (s *Server) StoreHeartbeat(ctx context.Context, request *pdpb.StoreHeartbea
 
 const regionHeartbeatSendTimeout = 5 * time.Second
 
+var heartbeatStreamRebindInterval = time.Minute
```
Comment (siddontang) [inline server/grpc_service.go]: can we use a config here to speed up test? 

I see that you should wait at least 1m in test to rebind.

Comment (disksing) [reply server/grpc_service.go]: I've set it to 3 seconds in test.

Comment (siddontang) [reply server/grpc_service.go]: oh, this is a var I just notice. Seem it is a not good way here.

**Comment by disksing** [comment]
PTAL @siddontang

**Comment by siddontang** [comment]
But CI failed @disksing

**Comment by disksing** [comment]
```
listen tcp 127.0.0.1:41210: bind: address already in use
```
I'm going to just restart it.

**Pull request merged**